src/libcore/result.rs

```diff
@@ -501,6 +501,8 @@ impl<T, E> Result<T, E> {
 
     /// Returns an iterator over the possibly contained value.
     ///
+    /// The iterator yields one value if the result is [`Ok`], otherwise none.
```

Comment:
Ok, I see your point. Confusion on my side. Sorry about that... Therefore, no change is need.